hudi-common/src/main/java/org/apache/hudi/common/table/timeline/dto/ClusteringOpDTO.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.common.table.timeline.dto;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.hudi.common.model.HoodieFileGroupId;
+import org.apache.hudi.common.table.timeline.HoodieInstant;
+import org.apache.hudi.common.util.collection.Pair;
+
+/**
+ * The data transfer object of compaction.
+ */
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class ClusteringOpDTO {
+
+  @JsonProperty("id")
+  private String fileId;
```

Comment:
What does a fileID mean here ? Is this the lowest representation of the clustering plan, will the clustering plan contain a bunch of ClusteringOpDTO's ?

Reply:
This is only used for RemoteFileSystemView implementation. RemoteFSV uses DTOs to serialize and deserialize the data over wire. One of the APIs filesystem view provides is to list all fileIds that have pending clustering scheduled. So this file contains all the information we need to send over network. It has nothing to do with clustering plan stored in avro format. Its just there to support getFilesInPendingClusteringOperation API

Reply:
Thanks for the explanation.